File: WindowsDesktop/EditRoom/DirtyAsteriskConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace EditRoom
{
    [ValueConversion(typeof(bool), typeof(string))]
    public class DirtyAsteriskConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            bool dirty = (bool)value;

            if (dirty)
                return "*";

            return string.Empty;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: WindowsMetro/EditRoom/DirtyAsteriskConverter.cpp
#include "pch.h"
#include "DirtyAsteriskConverter.h"

using namespace EditRoom;

static Platform::String^ asterisk("*");

DirtyAsteriskConverter::DirtyAsteriskConverter(void)
{
}


DirtyAsteriskConverter::~DirtyAsteriskConverter(void)
{
}

Platform::Object^ DirtyAsteriskConverter::Convert(Platform::Object^ value, Windows::UI::Xaml::Interop::TypeName targetType, Platform::Object^ parameter, Platform::String^ language)
{
	if(static_cast<Platform::Boolean>(value))
	{
		return asterisk;
	}
	return "";
}

Platform::Object^ DirtyAsteriskConverter::ConvertBack(Platform::Object^ value, Windows::UI::Xaml::Interop::TypeName targetType, Platform::Object^ parameter, Platform::String^ language)
{
	return static_cast<Platform::String^>(value) == asterisk;
}

